huggingface_pipelines/text.py

```diff
@@ -0,0 +1,355 @@
+from sonar.inference_pipelines.text import TextToEmbeddingModelPipeline, EmbeddingToTextModelPipeline
+import logging
+from typing import List, Dict, Any
+from .pipeline import Pipeline, PipelineOverwrites, PipelineConfig, PipelineFactory
+import torch
+from dataclasses import dataclass, replace
+import numpy as np
+from .dataset import DatasetConfig
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class TextToEmbeddingOverwrites(PipelineOverwrites, total=False):
```

Comment:
I think if you clean all of these Overwrites and `with_overwrite()` and use fairseq2's `update_dataclass` instead, this will free you up from writing many redundant classes and functions.